ingestr/src/destinations.py

```diff
@@ -261,3 +264,70 @@ def dlt_run_params(self, uri: str, table: str, **kwargs) -> dict:
 
     def post_load(self):
         pass
+
+
+class ClickhouseDestination:
+    def dlt_dest(self, uri: str, **kwargs):
+        parsed_uri = urlparse(uri)
+        print(f"parsed_uri: {parsed_uri}")
+
+        if "dest_table" in kwargs:
+            table = kwargs["dest_table"]
+            database = table.split(".")[0]
+        else:
+            database = parsed_uri.path.lstrip("/")
+
+        username = parsed_uri.username
+        if not username:
+            raise ValueError(
+                "A username is required to connect to the ClickHouse database."
+            )
+
+        password = parsed_uri.password
+        if not password:
+            raise ValueError(
+                "A password is required to authenticate with the ClickHouse database."
+            )
+
+        host = parsed_uri.hostname
+        if not host:
+            raise ValueError(
+                "The hostname or IP address of the ClickHouse server is required to establish a connection."
+            )
+
+        port = parsed_uri.port
+        if not port:
+            raise ValueError(
+                "The TCP port of the ClickHouse server is required to establish a connection."
+            )
+
+        http_port = (
+            int(parsed_uri.query.split("http_port=")[1])
+            if "http_port=" in parsed_uri.query
+            else 8123
+        )
```

Comment:
this logic is wrong. I think this would fail if the uri was like:
```
clickhouse://user:pw@host:port?http_port=1234&someotherkey=someothervalue
```

which would try to get `1234&someotherkey=someothervalue` and try to cast it to integer.

do it like this instead:
```suggestion
        http_port = (
            int(urlparse.parse_qs(parsed_uri.query)['http_port'][0])
            if "http_port=" in parsed_uri.query
            else 8123
        )
```